analytics2/utils/block_stats.py

```diff
@@ -217,7 +285,10 @@ def make_aoi_summary(bldg_pop_data: Union[str, gpd.GeoDataFrame],
     else:
         bldg_pop = load_bldg_pop(bldg_pop_data)
     block = flex_load(block_data)
+    bldg_pop = add_block_gini(bldg_pop, 'bldg_pop', block=block)
```

Comment:
since we do the same checks in each of these functions and the result of the failed check is similar to line 287, i think it makes sense to remove the call to `flex_load` within the bodies of these functions and just enforce the preconditions before we start adding features

same thing for the missing weights!